src/app/directives/datatable/column.directive.ts

```diff
@@ -0,0 +1,16 @@
+import { Directive, Host, Input, OnChanges } from "@angular/core";
+import { DataTableColumnDirective } from "@swimlane/ngx-datatable";
+
+@Directive({
+  // eslint-disable-next-line @angular-eslint/directive-selector
+  selector: "[sortKey]",
+})
+export class DatatableColumnDirective implements OnChanges {
+  @Input() public sortKey: string;
+
+  public constructor(@Host() private datatable: DataTableColumnDirective) {}
+
+  public ngOnChanges(): void {
+    this.datatable["sortKey"] = this.sortKey;
```

Comment:
nit: field name should not be `datatable`. I'd usually ignore this except that datatable is a valid parent component in the structure of the table. Should be `column` for short, or `columnDirective` .